pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -795,34 +795,33 @@ func setTrackerLabel(meta metav1.ObjectMeta, instance *v2vv1.VirtualMachineImpor
 func (r *ReconcileVirtualMachineImport) startVM(provider provider.Provider, instance *v2vv1.VirtualMachineImport, vmName types.NamespacedName) error {
 	vmi := &kubevirtv1.VirtualMachineInstance{}
 	err := r.client.Get(context.TODO(), vmName, vmi)
-	if err != nil {
-		if k8serrors.IsNotFound(err) {
-			if err = r.updateProgress(instance, progressStartVM); err != nil {
+	if err != nil && k8serrors.IsNotFound(err) {
+		if err = r.updateProgress(instance, progressStartVM); err != nil {
+			return err
+		}
+		if err = r.updateToRunning(vmName); err != nil {
+			// Emit event vm failed to start:
+			r.recorder.Eventf(instance, corev1.EventTypeWarning, EventVMStartFailed, "Virtual Machine %s/%s failed to start: %s", vmName.Namespace, vmName.Name, err)
```

Comment:
you can use `utils.ToLoggableResourceName(vmiName.Name, &vmiName.Namespace)`

Reply:
sure